src/core/mac/mac_frame.hpp

```diff
@@ -1545,14 +1568,42 @@ class BeaconPayload
      */
     bool IsJoiningPermitted(void) const { return (mFlags & kJoiningFlag) != 0; }
 
+    /**
+     * This method clears the Joining Permitted flag.
+     *
+     */
+    void ClearJoiningPermitted(void) { mFlags &= ~kJoiningFlag; }
+
+    /**
+     * This method sets the Joining Permitted flag.
+     *
+     */
+    void SetJoiningPermitted(void)
+    {
+        mFlags |= kJoiningFlag;
+
+#if OPENTHREAD_CONFIG_MAC_JOIN_BEACON_VERSION != 2 // check against kProtocolVersion
+        mFlags &= ~kVersionMask;
+        mFlags |= OPENTHREAD_CONFIG_MAC_JOIN_BEACON_VERSION << kVersionOffset;
+#endif
+    }
+
     /**
      * This method gets the Network Name field.
      *
-     * @returns The Network Name field as `NameData`.
+     * @returns The Network Name field as a string.
      *
      */
     const char *GetNetworkName(void) const { return mNetworkName; }
 
+    /**
+     * This method sets the Network Name field.
+     *
+     * @param[in]  aNetworkName  The Network Name as a string.
+     *
+     */
+    void SetNetworkName(const char *aNetworkName) { memcpy(mNetworkName, aNetworkName, sizeof(mNetworkName)); }
+
```

Comment:
Checking the implementation before 
- https://github.com/openthread/openthread/pull/7472

we did use `NameData` here which would take care of all these situations (null termination, etc) and is safe to use.

Suggest using the same model as it was before. Here is the link to the code right before the PR above was merged:

https://github.com/openthread/openthread/blob/a8b4fbb1893823e1c247ee1cc7dac8f61519cba4/src/core/mac/mac_frame.hpp#L1593-L1638